pkg/sdk/validation/policy/policy.go

```diff
@@ -95,8 +95,31 @@ func (v *Validator) ValidateAdditionalInputParameters(ctx context.Context, param
 
 // ValidateTypeInstancesMetadata validates that every TypeInstance has metadata resolved.
 func (v *Validator) ValidateTypeInstancesMetadata(in policy.Policy) validation.Result {
+	resultBldr := validation.NewResultBuilder("Metadata for")
+
+	unresolvedIDs := map[string]struct{}{}
 	unresolvedTypeInstances := metadata.TypeInstanceIDsWithUnresolvedMetadataForPolicy(in)
-	return v.validationResultForTIMetadata(unresolvedTypeInstances)
+	for _, ti := range unresolvedTypeInstances {
+		resultBldr.ReportIssue(string(ti.Kind), "missing Type reference for %s", ti.String(false))
+		unresolvedIDs[ti.ID] = struct{}{}
+	}
```

Comment:
I know the reason why you duplicated the logic of `validationResultForTIMetadata`, but maybe it would be better to change `validationResultForTIMetadata` to output both validation result and unresolvedIDs? In that way we won't have duplication.